src/integrationTest/java/com/mongodb/hibernate/BasicCRUDTests.java

```diff
@@ -141,33 +142,61 @@ void testEntityWithEmbeddedFieldInsertion() {
         }
     }
 
-    @Nested
-    class DeleteTests {
-
-        @Test
-        void testSimpleDeletion() {
-
-            // given
-            var id = 1;
-            sessionFactoryScope.inTransaction(session -> {
-                var book = new Book();
-                book.id = id;
-                book.title = "War and Peace";
-                book.author = "Leo Tolstoy";
-                book.publishYear = 1867;
-                session.persist(book);
-            });
-            assertThat(getCollectionDocuments()).hasSize(1);
-
-            // when
-            sessionFactoryScope.inTransaction(session -> {
-                var book = session.getReference(Book.class, id);
-                session.remove(book);
-            });
+    @Test
+    void testSimpleDeletion() {
+
+        // given
+        var id = 1;
+        sessionFactoryScope.inTransaction(session -> {
+            var book = new Book();
+            book.id = id;
+            book.title = "War and Peace";
+            book.author = "Leo Tolstoy";
+            book.publishYear = 1867;
+            session.persist(book);
+        });
+        assertThat(getCollectionDocuments()).hasSize(1);
+
+        // when
+        sessionFactoryScope.inTransaction(session -> {
+            var book = session.getReference(Book.class, id);
+            session.remove(book);
+        });
+
+        // then
+        assertThat(getCollectionDocuments()).isEmpty();
+    }
 
-            // then
-            assertThat(getCollectionDocuments()).isEmpty();
-        }
+    @Test
+    void testSimpleUpdating() {
+        var statementInspector = sessionFactoryScope.getStatementInspector(SQLStatementInspector.class);
+        statementInspector.clear();
+        sessionFactoryScope.inTransaction(session -> {
+            var book = new Book();
+            book.id = 1;
+            book.title = "War and Peace";
+            book.author = "Leo Tolstoy";
+            book.publishYear = 1867;
+            session.persist(book);
+            session.flush();
+
+            book.title = "Insurrection";
+            book.publishYear = 1899;
+        });
```

Comment:
from Hibernate javadoc, you might see 'merge' method is recommended in preference to `update`; but merge usage will require other translation logic. Not sure whether it should be in the scope of this ticket. If the update is the last action within a session, nothing needs to be done further.